types/npm-packlist/index.d.ts

```diff
@@ -1,18 +1,22 @@
 // Type definitions for npm-packlist 3.0
 // Project: https://github.com/npm/npm-packlist, https://www.npmjs.com/package/npm-packlist
 // Definitions by: Klaus Meinhardt <https://github.com/ajafff>
+//                 Florian Imdahl <https://github.com/ffflorian>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.1
+// Minimum TypeScript Version: 4.3
 
-declare function packlist(options?: packlist.Options): Promise<string[]>;
-declare function packlist<T>(options: packlist.Options | undefined, callback: (result: string[]) => T): Promise<T>;
+import * as Arborist from '@npmcli/arborist';
+import { Walker, WalkerSync } from 'ignore-walk';
+
+declare function packlist(tree: Arborist.Node, options?: packlist.Options): Promise<string[]>;
+declare function packlist<T>(tree: Arborist.Node, options: packlist.Options | undefined, callback: (result: string[]) => T): Promise<T>;
 declare namespace packlist {
     interface Options {
+        ignoreFiles?: string[] | undefined;
+        isSymbolicLink?: boolean | undefined;
```

Comment:
where does this option come from?

Reply:
The readme [stats](https://github.com/npm/npm-packlist/blob/main/README.md?plain=1#L104) that the options are inherited from [ignore-walk](https://github.com/npm/ignore-walk) and `isSymbolicLink` is used here: https://github.com/npm/ignore-walk/blob/main/lib/index.js#L13.

Reply:
ok, so should those be rather:
```ts
import type { Walker, WalkerSync, WalkerOptions } from 'ignore-walk';
```
```ts
declare namespace packlist {
    interface Options extends WalkerOptions {
        parent?: Walker | WalkerSync | null | undefined;
        /** Directory to walk recusively. Defaults to `process.cwd()`. */
        path?: string | undefined;
    }
....
```
?

Reply:
Yes, good idea.